The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10820886542/job/30021834827) [(plain)](https://github.com/rust-lang/rust/commit/f74b116e9024e62306ccb8bef0b137b1c7ccf83e/checks/30021834827/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.834 Building wheels for collected packages: reuse
#16 2.835   Building wheel for reuse (pyproject.toml): started
#16 3.079   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 3.080   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 3.080   Stored in directory: /tmp/pip-ephem-wheel-cache-e74dmxpp/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 3.083 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.483 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.484 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.6s
---
    Checking encode_unicode v0.3.6
error[E0599]: no variant or associated item named `is_async` found for enum `rustc_ast::CoroutineKind` in the current scope
   --> src/tools/clippy/clippy_utils/src/ast_utils.rs:224:52
    |
224 |                 && la.map_or(false, CoroutineKind::is_async) == ra.map_or(false, CoroutineKind::is_async)
    |                                                    ^^^^^^^^ variant or associated item not found in `CoroutineKind`
error[E0599]: no variant or associated item named `is_async` found for enum `rustc_ast::CoroutineKind` in the current scope
   --> src/tools/clippy/clippy_utils/src/ast_utils.rs:224:97
    |
    |
224 |                 && la.map_or(false, CoroutineKind::is_async) == ra.map_or(false, CoroutineKind::is_async)
    |                                                                                                 ^^^^^^^^ variant or associated item not found in `CoroutineKind`
    Checking windows-targets v0.48.5
    Checking quine-mc_cluskey v0.2.4
   Compiling anyhow v1.0.86
    Checking unicode-ident v1.0.12
```

</details>

--- a/compiler/rustc_ast/src/ast.rs
+++ b/compiler/rustc_ast/src/ast.rs
@@ -2610,14 +2610,6 @@ impl CoroutineKind {
         }
     }
 
-    pub fn is_async(self) -> bool {
-        matches!(self, CoroutineKind::Async { .. })
-    }
-
-    pub fn is_gen(self) -> bool {
-        matches!(self, CoroutineKind::Gen { .. })
-    }
-
     pub fn closure_id(self) -> NodeId {
         match self {
             CoroutineKind::Async { closure_id, .. }

--- a/src/tools/clippy/clippy_utils/src/ast_utils.rs
+++ b/src/tools/clippy/clippy_utils/src/ast_utils.rs
@@ -221,7 +221,7 @@ pub fn eq_expr(l: &Expr, r: &Expr) -> bool {
         ) => {
             eq_closure_binder(lb, rb)
                 && lc == rc
-                && la.map_or(false, CoroutineKind::is_async) == ra.map_or(false, CoroutineKind::is_async)
+                && eq_coroutine_kind(*la, *ra)
                 && lm == rm
                 && eq_fn_decl(lf, rf)
                 && eq_expr(le, re)
@@ -237,6 +237,16 @@ pub fn eq_expr(l: &Expr, r: &Expr) -> bool {
                 && eq_struct_rest(&lse.rest, &rse.rest)
                 && unordered_over(&lse.fields, &rse.fields, eq_field)
         },
+        _ => false,
+    }
+}
+
+fn eq_coroutine_kind(a: Option<CoroutineKind>, b: Option<CoroutineKind>) -> bool {
+    match (a, b) {
+        (Some(CoroutineKind::Async { .. }), Some(CoroutineKind::Async { .. }))
+        | (Some(CoroutineKind::Gen { .. }), Some(CoroutineKind::Gen { .. }))
+        | (Some(CoroutineKind::AsyncGen { .. }), Some(CoroutineKind::AsyncGen { .. }))
+        | (None, None) => true,
         _ => false,
     }
 }

